main.py

```diff
@@ -29,25 +30,38 @@ def process_email(email_path):
         elif k.lower() == "n":
             logging.info("---------Stop processing emails---------")
             return
+    email_counter = 0  
     for mail in mail_info:
         state, data = chatbot.get_content(mail)
         if state == 'Succeed':
             res.append(data)
-    return res
+            email_counter += 1
+            if email_counter % AUTO_SAVE_EMAIL == 0:
+                export_to_csv(res, output_csv)
+                res = []
+    return res, email_counter
 
 def export_to_csv(data, filename):
-    with open(filename, mode='w', newline='', encoding='utf-8') as csvfile:
+    file_exists = False
+    try:
+        with open(filename, 'r', newline='', encoding='utf-8') as csvfile:
+            file_exists = True
+    except FileNotFoundError:
+        pass
+    mode = 'a' if file_exists else 'w'
+    with open(filename, mode=mode, newline='', encoding='utf-8') as csvfile:
         fieldnames = data[0].keys()
         writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
-        writer.writeheader()
+        if not file_exists:
+            writer.writeheader()
         for row in data:
             writer.writerow(row)
 
 def main(email_path, output_csv):
-    result = process_email(email_path)
-    if result:
+    result, count = process_email(email_path, output_csv)
+    if count > 0:
         export_to_csv(result, output_csv)
-        logging.info(f"Processed emails successfully and exported to CSV at {output_csv}.")
+        logging.info(f"Processed {str(count)} emails successfully and exported to CSV at {output_csv}.")
```

Comment:
We should put this logging inside the `export_to_csv` function. So everytime we write to the CSV, we can display the log